force-app/main/default/classes/RD2_EntryFormController.cls

```diff
@@ -164,6 +164,36 @@ public with sharing class RD2_EntryFormController {
         };
     }
 
+    /**
+    * @description Retrieve the donor address given an Account or Contact Id
+    * @param donorId Id of donor, Account or Contact
+    * @return Map<String, String> of address values
+    */
+    @AuraEnabled
+    public static Address__c getDonorAddress(Id donorId) {
```

Comment:
This method needs to check Read access and field visibility on the Account/Contact address fields.